### PR TITLE
Check for divide-by-zero before calculating the initial velocity of spring animations.

### PR DESCRIPTION
**Pull request by jverkoey**
This improves the robustness of MDMConfigureAnimation while enabling us to call the method without having to check whether the from/to value are equal.

### CODE REVIEW

**Review by samnm — APPROVED** [review]

File: src/private/CABasicAnimation+MotionAnimator.m [hunk 1]
```diff
@@ -211,7 +222,9 @@ void MDMConfigureAnimation(CABasicAnimation *animation, MDMMotionTiming timing)
       CGFloat displacement = -biggestDelta;
       CGFloat absoluteInitialVelocity =
           timing.curve.data[MDMSpringMotionCurveDataIndexInitialVelocity];
-      springAnimation.initialVelocity = absoluteInitialVelocity / displacement;
+      if (fabs(displacement) > CGFLOAT_EPSILON) {
```
Comment (romoore) [inline src/private/CABasicAnimation+MotionAnimator.m]: Please consider just checking that it's greater than some sensible constant, like 0.00001.  If I understand correctly, FLT_EPSILON is not intended to be used in this manner (it's not the smallest non-zero positive value).  I think using a 0.000001 or similar would be more straightforward to maintainers who do not need to go and investigate the complexities of floating point epsilon values.

https://stackoverflow.com/questions/10334688/how-dangerous-is-it-to-compare-floating-point-values

Comment (romoore) [reply src/private/CABasicAnimation+MotionAnimator.m]: Another good read, forwarded by @willlarche 

https://www.jessesquires.com/blog/floating-point-swift-ulp-and-epsilon/

Comment (jverkoey) [reply src/private/CABasicAnimation+MotionAnimator.m]: Done.

File: src/private/CABasicAnimation+MotionAnimator.m [hunk 2]
```diff
@@ -180,7 +181,9 @@ void MDMConfigureAnimation(CABasicAnimation *animation, MDMMotionTiming timing)
       // As for our sign, if absoluteInitialVelocity matches the direction of displacement, then our
       // sign will be positive. Otherwise, our sign will be negative, as expected by Core Animation.
 
-      springAnimation.initialVelocity = absoluteInitialVelocity / displacement;
+      if (fabs(displacement) > 0.00001) {
```
Comment (romoore) [inline src/private/CABasicAnimation+MotionAnimator.m:183]: I don't think tgmath.h is required if you're going to (potentially) promote `displacement` to a double and then check the double return value.

**Review by romoore — APPROVED** [review]

File: src/private/CABasicAnimation+MotionAnimator.m (hunk 2 above)
Comment (jverkoey) [reply src/private/CABasicAnimation+MotionAnimator.m:183]: Done.

**Pull request merged**